frontend/src/components/custom-tools/tool-ide/ToolIde.jsx

```diff
@@ -206,14 +213,15 @@ function ToolIde() {
     const body = {
       output: doc?.document_id,
     };
-
-    handleUpdateTool(body).catch((err) => {
-      const revertSelectedDoc = {
-        selectedDoc: prevSelectedDoc,
-      };
-      updateCustomTool(revertSelectedDoc);
-      setAlertDetails(handleException(err, "Failed to select the document"));
-    });
+    if (!isPublicSource) {
+      handleUpdateTool(body).catch((err) => {
+        const revertSelectedDoc = {
+          selectedDoc: prevSelectedDoc,
+        };
+        updateCustomTool(revertSelectedDoc);
+        setAlertDetails(handleException(err, "Failed to select the document"));
+      });
+    }
```

Comment:
@harini-venkataraman Instead of having the update tool function call inside the condition, a better approach would be to return if it's a public source on line 212 of the same function.